src/main.ts

```diff
@@ -0,0 +1,171 @@
+import * as core from '@actions/core';
+import * as github from '@actions/github';
+import * as fs from 'fs';
+import { request } from '@octokit/request';
+import { Octokit } from '@octokit/core';
+
+async function main() { 
+  fs.readdir('./', (err, files) => {
+    if (err){
+      console.log(err);
+    }
+    else{
+      var secret_token = core.getInput('GITHUB_TOKEN');
+      const octokit = new Octokit({
+      auth: secret_token,
+      });
+      const repository = github.context.repo.repo;
+      const ownername = github.context.repo.owner;
+      //Check for example and Contribution in README
+      readmeChecks(files);
+      //Check for CODEOWNERS file in .github folder
+      codeOwnerCheck();
+      //Check if nodemodules folder is present in master branch for typescript action
+      nodeModulesCheck();
+      //check for branch permissions in main
+      branchPermissionCheck('main', repository, ownername, secret_token, octokit);
```

Comment:
Is there a possibility that some of the actions repo might have "master" instead of main?
It will be good to investigate a bit on this.